app/main/views/suppliers.py

```diff
@@ -119,6 +119,38 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/agreements/<agreement_id>/on-hold', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def put_signed_agreement_on_hold(agreement_id):
+    data_api_client.put_signed_agreement_on_hold(agreement_id, current_user.email_address)
+
+    organisation = request.args['organisation']
+    flash('The agreement for {} was put on hold.'.format(organisation), 'message')
```

Comment:
The flashed messages don't appear on the list page, but should be fine to leave it for when the "next agreement" flow gets written.